itwm_example/mco/mco.py

```diff
@@ -1,38 +1,45 @@
-import sys
-import subprocess
 import logging
+import sys
+
 import numpy as np
 from scipy import optimize
 
 from traits.api import (
-    HasStrictTraits, List, Float, Str, Instance, Interface, provides
+    HasStrictTraits, List, Float, Instance
 )
 
 from force_bdss.api import (
-    BaseMCO, BaseMCOParameter, Workflow, DataValue
+    BaseMCO, BaseMCOParameter, DataValue
 )
+from force_bdss.mco.i_evaluator import IEvaluator
 
+from .subprocess_workflow_evaluator import SubprocessWorkflowEvaluator
 
 log = logging.getLogger(__name__)
 
 
 class MCO(BaseMCO):
 
-    def run(self, model):
+    def run(self, evaluator):
+
+        model = evaluator.mco_model
         parameters = model.parameters
         kpis = model.kpis
 
-        application = self.factory.plugin.application
-        workflow_file = application.workflow_file
         if model.evaluation_mode == "Subprocess":
-            single_point_evaluator = SubprocessSinglePointEvaluator(
-                sys.argv[0], workflow_file.path
+            # Here we create an instance of our WorkflowEvaluator subclass
+            # that allows for evaluation of a state in the workflow via calling
+            # force_bdss on a new subprocess running in 'evaluate' mode.
+            # Note: a BaseMCOCommunicator must be present to pass in parameter
+            # values and returning the KPI for a force_bdss run in 'evaluate'
+            # mode
+            single_point_evaluator = SubprocessWorkflowEvaluator(
+                workflow=evaluator.workflow,
+                workflow_filepath=evaluator.workflow_filepath,
```

Comment:
Didn't we change `workflow_filepath` to `workflowfile.path`? It's not related to the code here, rather how the `evaluator` is implemented.